specification/under_development/profiles/Adaptive_Profile.md

```diff
@@ -0,0 +1,977 @@
+# Adaptive Profile
+
+This profile defines a subset of the QIR specification to support a coherent set
+of functionalities and capabilities that might be offered by a quantum backend.
+Like all profile specifications, this document is primarily intended for
+[compiler backend](https://en.wikipedia.org/wiki/Compiler#Back_end) authors as
+well as contributors to the [targeting
+stage](../Compilation_And_Targeting.md#targeting) of the QIR compiler.
+
+The adaptive profile specifies supersets of the base profile that enables
+control flow based on mid-circuit measurements and classical computations
+within coherence times. A back-end can support this profile by supporting
+a minimum set of features beyond the base profile and can opt-in for
+features beyond that.
+This marks the first attempt at defining
+a profile greater in scope than the base profile but still limited when
+compared to the full QIR specification.
+
+Being a superset of the base profile means that these 3 *capabilities* must be implemented (a few restrictions are removed from the base profile):
+
+1. An adaptive profile program can execute a sequence of quantum instructions
+   that transform the quantum state.
+2. An adaptive profile program can measure any qubit used in the program and thus
+   a back-end must be able to measure the state of all qubits for use in program output generation.
+3. An adaptive profile program must produce one of the specified [output schemas](../output_schemas/).
+
+Minimum capability additions to the base profile an adaptive profile supporting back-end must be able to support:
+
+<!-- markdownlint-disable MD029 -->
+4. Conditional branching (the profile must support the `br` instruction assuming that the instruction only expresses non-loop control flow). Since support for the `br`
```

Comment:
nit: since branching already implies a condition, maybe this could start with just "Branching".